tests/test_request.py

```diff
@@ -191,3 +191,27 @@ def test_bad_url_parse():
             Mock(),
             Mock(),
         )
+
+
+def test_request_scope_raises_exception_when_no_asgi():
+    app = Sanic("no_asgi")
+
+    @app.get("/")
+    async def get(request):
+        return request.scope
+
+    _, response = app.test_client.get("/")
```

Comment:
You could even go back similar to what your previous test looked like and test for the exceptions:

```
request, _ = ...

with pytest.raises(NotImplementedError):
    request.scope
```